gcp/cluster.tf

```diff
@@ -18,12 +18,6 @@ resource "google_compute_resource_policy" "redpanda-rp" {
     availability_domain_count = var.ha ? max(3, var.nodes) : 1
   }
   count = var.ha ? 1 : 0
-  lifecycle {
-    precondition {
-      condition = var.ha && var.nodes <= 8
-      error_message = "GCP does not support resource policies with greater than 8 nodes"
-    }
-  }
```

Comment:
Why was this removed? Not sure I understand how this is related to making the region selection work generically for all three clouds.

Reply:
It's not, but it doesn't work with this condition in. We could do it as separate PRs, but figured a single catch-all "make stuff work" PR would be fine

Reply:
Got it.

Reply:
I'd like to understand why it suddenly stopped working though

Reply:
Let's file an issue on this one in particular. after reading the google_compute_resource_policy flow in clusters.tf and looking at the docs, I don't understand where this limitation of 8 nodes comes from

Reply:
The issue here is that GCP does not give you visibility nor control over which failure domain a resource has been placed into (https://issuetracker.google.com/issues/256993209?pli=1). So the only way that we can guarantee that specific nodes are in separate racks is to put them into entirely separate failure domains - basically one broker per failure domain, and we are limited by the number of failure domains (at the moment 8).

Reply:
Got it! This explanation would be worth adding as a comment.